engine/src/main/java/org/pentaho/di/pan/delegates/PanTransformationDelegate.java

```diff
@@ -0,0 +1,366 @@
+    trans.prepareExecution( arguments );
+    trans.startThreads();
+
+    // Wait for completion
+    trans.waitUntilFinished();
+
+    return trans.getResult();
+  }
+
+  /**
+   * Execute transformation remotely.
+   */
+  private Result executeRemotely( TransMeta transMeta,
+                                  TransExecutionConfiguration executionConfiguration ) throws KettleException {
+
+    log.logBasic( BaseMessages.getString( pkg, "PanTransformationDelegate.Log.ExecutingRemotely" ) );
+
+    if ( executionConfiguration.getRemoteServer() == null ) {
+      throw new KettleException( BaseMessages.getString( pkg, "PanTransformationDelegate.Error.NoRemoteServerSpecified" ) );
+    }
+
+    // Send transformation to slave server
+    String carteObjectId = Trans.sendToSlaveServer( transMeta, executionConfiguration, repository, MetaStoreConst.getDefaultMetastore() );
+
+    // Monitor remote transformation
+    monitorRemoteTransformation( transMeta, carteObjectId, executionConfiguration.getRemoteServer() );
+
+    // For command-line execution, we typically return a simple success result
+    // In a real implementation, you might want to fetch the actual result from the remote server
+    Result result = new Result();
+    result.setResult( true );
+    return result;
+  }
+
+  /**
+   * Execute transformation in clustered mode.
+   */
+  protected Result executeClustered( TransMeta transMeta,
+                                   TransExecutionConfiguration executionConfiguration ) throws KettleException {
+
+    log.logBasic( BaseMessages.getString( pkg, "PanTransformationDelegate.Log.ExecutingClustered" ) );
+
+    try {
+      final TransSplitter transSplitter = new TransSplitter( transMeta );
+      transSplitter.splitOriginalTransformation();
+
+      // Inject certain internal variables to make it more intuitive
+      for ( String transVar : Const.INTERNAL_TRANS_VARIABLES ) {
+        executionConfiguration.getVariables().put( transVar, transMeta.getVariable( transVar ) );
+      }
+
+      // Parameters override the variables
+      TransMeta originalTransformation = transSplitter.getOriginalTransformation();
+      for ( String param : originalTransformation.listParameters() ) {
+        String value = Const.NVL( originalTransformation.getParameterValue( param ),
+          Const.NVL( originalTransformation.getParameterDefault( param ),
+            originalTransformation.getVariable( param ) ) );
+        if ( !Utils.isEmpty( value ) ) {
+          executionConfiguration.getVariables().put( param, value );
+        }
+      }
+      executeClustered( transSplitter, executionConfiguration );
+      // Monitor clustered transformation
+      Trans.monitorClusteredTransformation( log, transSplitter, null );
+      Result result = Trans.getClusteredTransformationResult( log, transSplitter, null );
+
+      logClusteredResults( transMeta, result );
+
+      return result;
+
+    } catch ( Exception e ) {
```

Comment:
Why are we catching Exception and casting it to KettleException?
Rarely should you catch runtime Exceptions and cast them to checked exceptions.

I can see a case if this was how it was before.